src/components/GroupMembersList/groupActions.ts

```diff
@@ -0,0 +1,54 @@
+import {
+    GroupActionsTypes,
+    SET_GROUP,
+} from './groupActionTypes'
+import { Api } from '../../services/api'
+import { showErrorMessage } from '../App/Snackbar/snackbarActions'
+import { selectToken } from '../App/userSelector'
+import { selectGroup } from './groupSelector'
+import _ from 'lodash'
+
+
+export const setGroup = (group: any): GroupActionsTypes => ({
+    type: SET_GROUP,
+    payload: {
+        group,
+    },
+})
+
+export const getGroup = (t, groupId) => async (dispatch, getState) => {
+    const token = selectToken(getState())
+    const api = new Api(token)
+
+    try {
+        const group = await api.getGroup(groupId)
+        dispatch(setGroup(group))
+    } catch (err) {
+        dispatch(showErrorMessage(t('errors.group-fetch')))
+    }
+}
+
+export const deleteMembers = (t, groupId, members) => async (dispatch, getState) => {
+    const token = selectToken(getState())
+    const group = selectGroup(getState())
+    const api = new Api(token)
+
+    if(!group?.members){
+        return
+    }
+
+    const newGroup = {
+        ...group,
+        members :  _.differenceBy(group.members, members,'id')
+    }
+    
+    dispatch(setGroup(newGroup))
```

Comment:
i think you should `dispatch` after `api delete`

Reply:
I decided to do it before api delete for a better UX experience. Here the user can click on the 'X', the member will be deleted form the group state and be removed from the list. Then the api is called and if it fails an error will be displayed and the state of the group will rollback. (and the list will be refreshed).

I you prefere the other way I can change it. (click 'X', show loader and wait for the api result to remove the member UI side)

Reply:
it's ok ;)